src/static/js/webmentions.js

```diff
@@ -0,0 +1,288 @@
+          <strong class="webmention-author-name">${mention["author"]["name"]}</strong>
+        </a>
+        <div class="webmention-content">
+          ${mentionContent}
+        </div>
+        <div class="webmention-meta">
+          <time
+            class="webmention-pub-date"
+            datetime="${mention["published"]}"
+          >
+            ${formatDate(mention["published"])}
+          </time>
+          <span class="webmention-divider" aria-hidden="true">⋅</span>
+          <a
+            class="webmention-source"
+            href="${mention["url"]}"
+            aria-label="View source of mention by ${mention["author"]["name"]}"
+          >View Source</a>
+        </div>
+      </li>
+    `;
+    mentionsHtmlElements.push(mentionHtml);
+  })
+  document.querySelector("#mentions-panel").innerHTML = `
+    <ul class="webmention-mentions">
+      ${mentionsHtmlElements.join("\n")}
+    </ul>
+  `;
+}
+
+// Parses and renders mentions into likes, reposts, replies and mentions
+function renderWebmentions(webmentions) {
+  if (!webmentions.length) {
+    return;
+  }
+
+  const likes = parseMentions(webmentions, "like-of");
+  const reposts = parseMentions(webmentions, "repost-of");
+  const replies = parseMentions(webmentions, "in-reply-to");
+  const mentions = parseMentions(webmentions, "mention-of");
+
+  if (likes.length) {
+    renderLikes(likes);
+  }
+
+  if (reposts.length) {
+    renderReposts(reposts);
+  }
+
+  if (replies.length) {
+    renderReplies(replies);
+  }
+
+  if (mentions.length) {
+    renderMentions(mentions);
+  }
+}
+
+// Process webmention promise
+function processWebmentions(targetURL) {
+  getWebmentions(targetURL)
+    .then(webmentions => renderWebmentions(webmentions))
+    .catch(e => console.error(e))
+}
+
+// Change tabs for webmentions UI
+function changeTabs(e) {
+  const target = e.target;
+  const parent = target.parentNode;
+  const grandparent = parent.parentNode;
+
+  // Remove all current selected tabs
+  parent
+    .querySelectorAll('[aria-selected="true"]')
+    .forEach(t => t.setAttribute("aria-selected", false));
+
+  // Set this tab as selected
+  target.setAttribute("aria-selected", true);
+
+  // Hide all tab panels
+  grandparent
+    .querySelectorAll('[role="tabpanel"]')
+    .forEach(p => p.setAttribute("hidden", true));
+
+  // Show the selected panel
+  grandparent.parentNode
+    .querySelector(`#${target.getAttribute("aria-controls")}`)
+    .removeAttribute("hidden");
+}
+
+function addTabListeners() {
+  const tabs = document.querySelectorAll('.reactions [role="tab"]');
+  const tabList = document.querySelector('.reactions [role="tablist"]');
+
+  // Add a click event handler to each tab
+  tabs.forEach(tab => {
+    tab.addEventListener("click", changeTabs);
+  });
+
+  // Enable arrow navigation between tabs in the tab list
+  let tabFocus = 0;
+
+  tabList.addEventListener("keydown", e => {
+    if (e.keyCode === 39 || e.keyCode === 37) {
```

Comment:
Can we use the key names for these codes, or set them to constants so it's clear what they represent?